pkg/component/controller/apiendpointreconciler_test.go

```diff
@@ -66,14 +65,13 @@ func TestBasicReconcilerWithNoLeader(t *testing.T) {
 
 func TestBasicReconcilerWithNoExistingEndpoint(t *testing.T) {
 	fakeFactory := testutil.NewFakeClientFactory()
-	config := &v1beta1.ClusterConfig{
-		Spec: &v1beta1.ClusterSpec{
-			API: &v1beta1.APISpec{
-				Address:         "1.2.3.4",
-				ExternalAddress: "get.k0s.sh",
-			},
-		},
-	}
+	config := getFakeConfig()
+
+	srv, err := getMockDNS()
+	require.NoError(t, err)
+
+	srv.PatchNet(net.DefaultResolver)
```

Comment:
I'd probably chosen a more lightweight solution, by having an interface with the LookupIPAddr function and an extra field of that interface type in the component struct that will be set to a dummy implementation in the tests. So we spare that extra dependency.

Reply:
I'll do that, has the advantage of making possible to run tests in parallel in the future and not adding the dependency is an advantage.